### DISCUSSION TITLE
Searching logs text in google - cleaning for special chracters and trimming the resultant search string

### DISCUSSION THREAD

**Discussion by Mohkam** [Pitmairen/selection-search#147] [Q&A] [answered]
Dear @Pitmairen ,

I often need to search for the log messages or the stack trace of different programming languages. it consume time to clean up those logs/traces of special characters like:  ~`!@#$%^&*()_+=-{}[]().,/\:;"'|

so i found that using regex: {%s|replace:/[\W_]/g/ /g} 
with below input:
"abc@#$%!@#$%^&**()_+{}|:<>?~`"def"ghi"
results in: 
 abc                           def ghi  
which is expected.

how can i trim so it replace multiple consecutive spaces with single ? so result may look like:
abc def ghi
is it even supported at the moment ?

thank you so much for this awesome extension.

**Comment by Pitmairen** [comment, marked as answer]
Hi, you can chain multiple replacements together, so something like this should work:

`{%s|replace:/[\W_]/ /g|replace:/\s+/ /g}`

The first replaces the special characters, and the second one removes any consecutive white space.

**Reply by Mohkam** [reply]
Thank you so much  ! it worked like a charm.
if you find it worthy, please do add this or similar example of multiple replacement in your official example documents.
Kind Regards,
Mohkafm

**Reply by Pitmairen** [reply]
Will do :)
